Documentation/requirements.md

```diff
@@ -0,0 +1,84 @@
+# Requirements
+
+For the most part the codebase is self-contained (e.g. all dependencies are vendored), but assembly of the stage1 requires some other tools to be installed on the system.
+
+## Build-time Requirements
+
+### Basic
+
+* GNU Make
+* Go 1.4+ (ideally 1.5.3 or later)
+* autoconf
+* aclocal (usually a part of automake)
+* bash
+* git
+* glibc
+  * development headers
+  * the rkt binary links against the library
+* gofmt (usually distributed with Go)
+* govet (usually distributed with Go)
+* TrouSerS (only when TPM is enabled)
+  * development headers
+  * the rkt binary links against the library
+* gpg (when running functional tests)
+
+### Additional requirements when also building any stage1 image flavor
+
+* glibc
+  * development headers
+  * the stage1 binaries link against the static library
+* libdl
+  * development headers
+  * the stage1 binaries link against the library
+* libacl
+  * development headers
+* C compiler
+
+### Specific for the coreos/kvm flavor
+
+* cat
+* cpio
+* gzip
+* md5sum
+* mktemp
+* sort
+* unsquashfs
+* wget
+* gpg (optional, required when downloading the CoreOS PXE image during the build)
+
+### Specific for the kvm flavor
+
+* patch
+* tar
+* xz
+* build requirements for kernel
+* build requirements for lkvm
+
+### Specific for the src flavor
+
+* build requirements for systemd
+
+## Run-time requirements
+
+* Linux 3.8+ (ideally 4.3+ to have overlay-on-overlay working), with the following options configured:
+  * CONFIG_CGROUPS
+  * CONFIG_NAMESPACES
+  * CONFIG_UTS_NS
+  * CONFIG_IPC_NS
+  * CONFIG_PID_NS
+  * CONFIG_NET_NS
+  * CONFIG_OVERLAY_FS (nice to have)
+
+### Additional run-time requirement for all the stage1 image flavors
+
+* libacl
+  * the library is optional (it is dlopened inside the stage1)
```

Comment:
Is this really a "requirement" if it is optional? Should the heading above therefore say something like, "Additonal run-time linkage for all stage1 flavors"?